checker/src/test/java/com/pushtorefresh/private_constructor_checker/PrivateConstructionCheckerTest.java

```diff
@@ -1,75 +1,55 @@
 package com.pushtorefresh.private_constructor_checker;
 
+import org.junit.Rule;
 import org.junit.Test;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import org.junit.rules.ExpectedException;
 
 public class PrivateConstructionCheckerTest {
+    @Rule public ExpectedException expectedException = ExpectedException.none();
 
     @Test
     public void builderShouldThrowExceptionIfNullWasPassedAsExpectedTypeOfException() {
-        try {
-            PrivateConstructorChecker
-                    .forClass(Object.class)
-                    .expectedTypeOfException(null);
-
-            fail();
-        } catch (IllegalArgumentException expected) {
-            assertEquals("expectedTypeOfException can not be null", expected.getMessage());
-        }
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("expectedTypeOfException can not be null");
+        PrivateConstructorChecker
+                .forClass(Object.class)
+                .expectedTypeOfException(null);
```

Comment:
Can you please divide `forClass` and `expectedTypeOfException`, I mean, to make sure that it fails after call to the `expectedTypeOfException` and not to the `forClass`